packages/store/tests/signal-per-state.spec.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  APP_INITIALIZER,
+  Component,
+  Injectable,
+  effect,
+  inject,
+  provideExperimentalZonelessChangeDetection,
+  signal
+} from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { Action, NgxsModule, Selector, State, StateContext, Store } from '@ngxs/store';
+import { skipConsoleLogging } from '@ngxs/store/internals/testing';
+
+describe('State per signal', () => {
+  class Set {
+    static readonly type = 'Set';
+
+    constructor(readonly value: number) {}
+  }
+
+  @State<number>({
+    name: 'number',
+    defaults: 0
+  })
+  @Injectable()
+  class NumberState {
+    @Selector()
+    static getNumber(number: number): number {
+      return number;
+    }
+
+    @Action(Set)
+    set(ctx: StateContext<number>, action: Set) {
+      ctx.setState(action.value);
+    }
+  }
+
+  @Injectable({ providedIn: 'root' })
+  class TestService {
+    number = inject(Store).selectSignal(NumberState.getNumber);
+
+    constructor() {
+      console.log(this.number());
+    }
+  }
+
+  @Component({
+    template: `
+      <h1>{{ value() }}</h1>
+      <h2>{{ number() }}</h2>
+      <button (click)="updateValue()">Click me</button>
+    `,
+    standalone: true
+  })
+  class TestComponent {
+    value = signal(0);
+    number = this.store.selectSignal(NumberState.getNumber);
+
+    constructor(private store: Store) {
+      effect(
+        () => {
+          const value = this.value();
+          store.dispatch(new Set(value));
```

Comment:
Maybe a silly comment, but calling this action `Set` can be confusing in terms of readability of this code. Because it can lead to confusion with the Set data structure of JavaScript. 

We could use `SetNumber`.

Please let me know if this kind of comment is irrelevant

Reply:
I will update its name